src/components/auth/ApiKeyAuth/ApiKeyAuthFlow.tsx

```diff
@@ -37,16 +38,21 @@ export function ApiKeyAuthFlow({
 
   const onNext = useCallback(
     (form: IFormType) => {
-      const { apiKey } = form;
+      const { apiKey, providerMetadata } = form;
+      const apiProviderMetadata = toApiProviderMetadata(providerMetadata);
+      const providerWorkspaceRef = apiProviderMetadata?.workspace?.value;
```

Comment:
We work with the workspace separately and as a part of the metadata both, and eventually can only have it be a part of the metadata.